service/switchboardapp/src/test/java/com/switchboard/app/DeviceControllerTest.java

```diff
@@ -0,0 +1,103 @@
+package com.switchboard.app;
+
+import com.switchboard.app.controller.DeviceController;
+import com.switchboard.app.dao.DeviceDaoImpl;
+import com.switchboard.app.domain.DeviceEntity;
+import com.switchboard.app.exceptions.DeviceAlreadyExistsException;
+import com.switchboard.app.exceptions.DeviceNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+class DeviceControllerTest {
+    @InjectMocks
+    DeviceController deviceController;
+
+    @Mock
+    DeviceDaoImpl deviceService;
+
+    private DeviceEntity device1;
+
+    @BeforeEach
+    void setUp(){
```

Comment:
Typo here, `setup`

Reply:
@FGRCL Where is the typo? setUp() is a junit method that he is overriding.

Reply:
I've changed it to setup